### PR TITLE
Comment tweaks

### LINKED ISSUE #632: Bot Comment Tweaks
1. Error messages should be written as a comment leveraging `diff` and not prefix "Error: " because "Error: " recurses.

```diff
- My error message
```

Markdown source code for reference:

Screenshot 2023-08-18 at 05 33 46

2. Permits valued at 0 should be skipped for commenting out or logged as debt. 

>> Error: Error: new row violates row-level security policy for table "permits"
> 
> Perhaps we should not prefix error messages with "Error: " or else nested errors look like this 
> 
>> ### [ **pavlovcik: [ CLAIM 0 WXDAI ]** ](https://pay.ubq.fi?claim=eyJwZXJtaXQiOnsicGVybWl0dGVkIjp7InRva2VuIjoiMHhlOTFEMTUzRTBiNDE1MThBMkNlOERkM0Q3OTQ0RmE4NjM0NjNhOTdkIiwiYW1vdW50IjoiMCJ9LCJub25jZSI6IjcwMDA5NTc5MTE3MjM0MzA5MTExOTE4NTYwNjI3MjU0MzI2MzAxODYzMjA3NDcyMjkwNjI2NTU3ODA2MTYyMjg2MTc0MjcwODA4OTk2IiwiZGVhZGxpbmUiOiIxMTU3OTIwODkyMzczMTYxOTU0MjM1NzA5ODUwMDg2ODc5MDc4NTMyNjk5ODQ2NjU2NDA1NjQwMzk0NTc1ODQwMDc5MTMxMjk2Mzk5MzUifSwidHJhbnNmZXJEZXRhaWxzIjp7InRvIjoiMHg0MDA3Q0UyMDgzYzdGM0UxODA5N2FlQjNBMzliYjhlQzE0OWEzNDFkIiwicmVxdWVzdGVkQW1vdW50IjoiMCJ9LCJvd25lciI6IjB4NzBmYmNGODJmZmE4OTFDNDI2N0I3Nzg0N2MyMTI0M2M1NjZmNzYxNyIsInNpZ25hdHVyZSI6IjB4ZGRkYTYzODIzNjVlMWQ0YTE0ZjhkZjJlMDA1NmJiODgwNjA4ZDhkNmMzOWYyNTMzNTgxMjhjNGE2MjUwYjEzZDEzNGRhYjA2YjA5ZjIwMzdlN2Y4YzJjYjdlMjk1MmE1MzE2ODhkYThiYTM5MmFiZWY1YzA2MjMzZWJiZTMwNjcxYyJ9&network=100)
>> 
> 
> Also 0 WXDAI permits we should not print out. 
> 
>> its payment of 0.0 WXDAI will be deducted from your next bounty.
> 
> Or appear as debt.

_Originally posted by @pavlovcik in https://github.com/ubiquibot/staging/issues/80#issuecomment-1676941586_

### PR DESCRIPTION
**Pull request by Keyrxng**
Resolves #632

Logging skipping due to zero penalty

```ErrorDiff(err)``` has replaced bot GitHub comment errors but not db logging

### CODE REVIEW

**Comment by Keyrxng** [comment]
What's the easiest way to force an error folks any ideas?

File: src/handlers/comment/handlers/index.ts [hunk 1]
```diff
@@ -176,20 +177,24 @@ export const issueReopenedCallback = async (): Promise<void> => {
...
     const assignee = events[0].assignee.login;
 
-    // write penalty to db
-    try {
-      await addPenalty(assignee, repository.full_name, tokenAddress, networkId.toString(), amount);
-    } catch (err) {
-      logger.error(`Error writing penalty to db: ${err}`);
-      return;
-    }
+    if (formattedAmount != "0.0") {
```
Comment (pavlovcik) [inline src/handlers/comment/handlers/index.ts:180]: I wonder if this is the most robust way to handle this. I don't have any other suggestions at this time.

File: src/handlers/comment/handlers/table.ts [hunk 2]
```diff
@@ -16,24 +16,23 @@ export const tableComment = ({
   days?: number;
 }) => {
   return `
-<code>
```
Comment (pavlovcik) [inline src/handlers/comment/handlers/table.ts:19]: This was intentional to display the table in monospace.

File: src/utils/helpers.ts [hunk 3]
```diff
@@ -20,3 +20,7 @@ export const getNumericLevel = (level: Level) => {
       return -1; // Invalid level
   }
 };
+
+export const ErrorDiff = (message: unknown) => {
+  return "```diff\n- " + message + "\n```";
```
Comment (pavlovcik) [inline src/utils/helpers.ts:25]: This is a neat implementation thanks for this. Would you happen to have an example comment of this error rendering?

**Review by pavlovcik — CHANGES_REQUESTED** [review]

File: src/utils/helpers.ts (hunk 3 above)
Comment (Keyrxng) [reply src/utils/helpers.ts:25]: I just replaced the tableComment with the ErrorDiff to demo it

https://github.com/Keyrxng/UbiquityBotTesting/issues/43#issuecomment-1712851989

File: src/handlers/comment/handlers/index.ts (hunk 1 above)
Comment (Keyrxng) [reply src/handlers/comment/handlers/index.ts:180]: I'll parse it as a float check for ``> 0``

File: src/handlers/comment/handlers/table.ts (hunk 2 above)
Comment (Keyrxng) [reply src/handlers/comment/handlers/table.ts:19]: https://github.com/Keyrxng/UbiquityBotTesting/issues/43#issuecomment-1712886879

Reverted it back and managed to remove some of the whitespace

**Review by pavlovcik — APPROVED** [review]

**Pull request merged**